src/extension.ts

```diff
@@ -233,7 +235,8 @@ export function activate(context: vscode.ExtensionContext): void {
         // prompt user to deploy to newly created web app
         if (await vscode.window.showInformationMessage('Deploy to deployment slot?', yesButton, noButton) === yesButton) {
             this.properties[deployingToDeploymentSlot] = 'true';
-            const fsPath = await util.showWorkspaceFoldersQuickPick("Select the folder to deploy", this.properties);
+            const subpath: string = vscode.workspace.getConfiguration(extensionPrefix).get(configurationSettings.deploySubpath);
```

Comment:
Hmm still not quite right. The second parameter to `get` is missing:
```
vscode.workspace.getConfiguration(extensionPrefix).get(configurationSettings.deploySubpath, **FOLDERPATH**)
```

This is important for multi-root workspaces where the setting might apply on a per-folder basis. Rather than passing in `subpath` to `showWorkspaceFoldersQuickPick`, you could just pass in `configurationSettings.deploySubpath`. Then you can get `subPath` inside `vscode.workspace.workspaceFolders.map`

Reply:
So... looking at the API documentation... there doesn't appear to be a way to do the folder path on the get: https://code.visualstudio.com/docs/extensionAPI/vscode-api#WorkspaceConfiguration... I believe that's something that's only part of the "Update" for configurations...

I did test this using single folders, multiple folders without a workspace, and using workspaces, and I don't remember running into an issue where I would need to specify a folder path on the get...

Reply:
Sorry - I meant to say 'getConfiguration', not 'get'

Here's more info about multi-root workspaces (if you're not familiar): https://code.visualstudio.com/docs/editor/multi-root-workspaces
Make sure you 'Add folder to workspace' as described

Reply:
Ok, I think I finally see what you mean... sorry to be argumentative... this was easier in the branch that I created for adding the subpath from the context menu...

I moved the get configuration to the `showWorkspaceFoldersQuickPick`, as requested. There's still one tiny issue though that I'm not sure how to fix... if you set the subpath setting in the folder settings, it works great... but if you set it in the workspace settings, then it still applies it to all the folders in the quick pick. I think I might be able to use the .inspect<T> method of getConfiguration to work around this... thoughts?

Reply:
I think that's by design - sometimes people want all of their workspace folders to have the same settings